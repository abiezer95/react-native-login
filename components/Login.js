
import React, { Component } from 'react';
import { StyleSheet,Image, Text } from 'react-native';
import { Container,
  Header,Right, Button,Icon,
  Content,Footer,Fab,Item,
  FooterTab,Card ,
  Input,Title, Form, Left} from 'native-base';
import { Link } from 'react-router-native';

export default class Login extends React.Component {
    constructor(props){
      super(props)
      this.state = {
        user: '',
        password: '',
        isLogined: false
      }
    }

    render() {
    return (
      <Container style={styles.Body}>
        
      <Header style={styles.paddingAdictional}>
      <Title >MedTegritas</Title>
      </Header>
        <Content>
          <Image style={styles.image_style} source={{uri:'https://www.lupusasturias.org/data/fotos/noticias/g_20_historia_clinica_.png'}} />
            <Form style={styles.information}>
            <Item>
            <Icon name='people' />
                <Input style ={styles.InputDesingExtra} placeholder="User-Name"/>
            </Item>
            <Item>
            <Icon name="key"/>
                <Input style ={styles.InputDesingExtra} placeholder="Password"/>
            </Item>                  
                <Button light style={styles.adaptationOfButton} onPress={()=>{alert('Bienvenido/a !')}}><Text style={styles.textBtn}>Log in</Text></Button>
            </Form>
            <Content>
              <Link to="/signup">
              <Text style={styles.textBtn}>Ya tienes Cuenta ?</Text>
             </Link>
             </Content>
        </Content>
      <Footer/>
    </Container>
    // Register
    
    
  );
}
}

const styles = StyleSheet.create({
Body:{
  backgroundColor:'#DCB2A9'
},
image_style:{
  marginLeft:'23%',
  marginTop:15,
  height: 200,
  width: "60%",
  flex: 1,
  borderRadius:500
},
information: {
  flex: 1,
  backgroundColor: '#FCFCF8',
  marginTop:'5%',
  margin:'5%',
  width:'90%',
  height:400,
  alignItems: 'center',
  borderRadius:10,
  justifyContent: 'center',
  padding:'2%',
  paddingTop:100,
  paddingBottom:'1%',
},
InputDesingExtra:{
  borderColor: 'transparent',
  borderWidth: 1 ,padding:5,width:'100%'
},
paddingAdictional:{
  paddingTop:30
},
adaptationOfButton:{
  width:'90%',
  justifyContent:'center',
  margin:20,
  
},
 textBtn:{
  color:'#000',
  padding:1,
  fontWeight:'bold',
  textAlign: 'center',
  // fontFamily: 'Cochin',
  
},


});
module.export = Login