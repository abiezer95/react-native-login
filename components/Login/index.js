import React, {Component} from 'react';
import {Image, StyleSheet, Text, TouchableOpacity} from 'react-native';
import {widthPercentageToDP as wp} from 'react-native-responsive-screen'
import {Button, Container, Content, Form, Icon, Input, Item} from 'native-base';


export default class Login extends Component {
  render() {
    return (
      <Container style={styles.Body}>
        <Content>
          <Image style={styles.image_style}
                 source={{uri: 'https://www.lupusasturias.org/data/fotos/noticias/g_20_historia_clinica_.png'}}/>
          <Form style={styles.information}>
            <Item>
              <Icon name='people'/>
              <Input style={styles.InputDesingExtra} placeholder="User-Name"/>
            </Item>
            <Item>
              <Icon name="key"/>
              <Input style={styles.InputDesingExtra} placeholder="Password"/>
            </Item>
            <Button light style={styles.adaptationOfButton} onPress={() => {
              alert('Bienvenido/a!')
            }}><Text style={styles.textBtn}>Log in</Text></Button>
          </Form>
          <TouchableOpacity light style={styles.adaptationOfButton} onPress={() => ref.current?.navigate('sign-up')}>
            <Text style={styles.textBtn}>Register</Text>
          </TouchableOpacity>
        </Content>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  Body: {
    backgroundColor: '#DCB2A9'
  },
  image_style: {
    flex: 1,
    height: wp('40%'),
    justifyContent: 'center',
    marginTop: 15,
    width: wp('40%'),
    marginHorizontal: wp('30%'),
  },
  information: {
    flex: 1,
    backgroundColor: '#FCFCF8',
    margin: '5%',
    width: wp('90%'),
    alignContent: 'center',
    borderRadius: 10,
    justifyContent: 'space-between',
    paddingHorizontal: '2%',
    paddingVertical: '10%'
  },
  InputDesingExtra: {
    borderColor: 'transparent',
    borderWidth: 1,
    padding: 5,
    width: '100%'
  },
  paddingAdditional: {
    paddingTop: 30
  },
  adaptationOfButton: {
    width: '90%',
    justifyContent: 'center',
    margin: 20,

  },
  textBtn: {
    color: '#000',
    padding: 1,
    fontWeight: 'bold',
    textAlign: 'center',
    // fontFamily: 'Cochin',

  },


});
module.export = Login