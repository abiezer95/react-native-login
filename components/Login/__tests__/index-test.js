import React from 'react';
import 'react-native';
import Login from '../index';
import renderer from 'react-test-renderer';
// import @react-native-community/async-storage;

test('renders correctly', () => {
  const tree = renderer.create(<Login/>).toJSON();
  expect(tree).toMatchSnapshot();
  });
  
 