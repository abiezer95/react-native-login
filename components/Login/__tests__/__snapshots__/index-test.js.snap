// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#DCB2A9",
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://www.lupusasturias.org/data/fotos/noticias/g_20_historia_clinica_.png",
          }
        }
        style={
          Object {
            "flex": 1,
            "height": 300,
            "justifyContent": "center",
            "marginHorizontal": 225,
            "marginTop": 15,
            "width": 300,
          }
        }
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "alignContent": "center",
              "backgroundColor": "#FCFCF8",
              "borderRadius": 10,
              "flex": 1,
              "justifyContent": "space-between",
              "margin": "5%",
              "paddingHorizontal": "2%",
              "paddingVertical": "10%",
              "width": 675,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRadius": undefined,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 15,
              "opacity": 1,
            }
          }
        >
          <Text />
          <TextInput
            allowFontScaling={true}
            editable={true}
            placeholder="User-Name"
            placeholderTextColor="#575757"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "top": 1.5,
                },
                Object {
                  "borderColor": "transparent",
                  "borderWidth": 1,
                  "padding": 5,
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="rgba(0,0,0,0)"
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRadius": undefined,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 15,
              "opacity": 1,
            }
          }
        >
          <Text />
          <TextInput
            allowFontScaling={true}
            editable={true}
            placeholder="Password"
            placeholderTextColor="#575757"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "top": 1.5,
                },
                Object {
                  "borderColor": "transparent",
                  "borderWidth": 1,
                  "padding": 5,
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="rgba(0,0,0,0)"
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f4f4f4",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontWeight": "bold",
                "padding": 1,
                "textAlign": "center",
              }
            }
          >
            Log in
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "margin": 20,
            "opacity": 1,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontWeight": "bold",
              "padding": 1,
              "textAlign": "center",
            }
          }
        >
          Register
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
